### PR TITLE
Issue 8

### LINKED ISSUE #8: support default values source
in this case you should realize parse default values from default tag in structures. For example:
```golang
type MyStruct struct {
     Field string `cf_default:"test"`
}
```

Need support all of this types: int8, int16, int32, int64, float16, float32, float64, string, bool, char, slice, map

### PR DESCRIPTION
**Pull request by kubitre**
resolve #8

### CODE REVIEW

**Comment by artemkaxboy** [comment]
This branch cannot be rebased due to conflicts

File: pipeline/environment_function_test.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+				},
+			},
+			want: infra.NewGoStructorTrueValue(reflect.ValueOf("tururu")),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			config := EnvironmentConfig{}
+			got := config.GetBaseType(tt.args.context)
+			os.Remove("testBaseType")
```
Comment (artemkaxboy) [inline pipeline/environment_function_test.go:46]: Use defer

File: pipeline/environment_function_test.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+			}
+		})
+	}
+}
+
+func TestEnvironmentConfig_GetComplexType(t *testing.T) {
+	myStruct := struct {
+		fieldhard []float32 `cf_env:"mySlice"`
+	}{}
+	os.Setenv("mySlice", "1.12,1.24,1.67")
```
Comment (artemkaxboy) [inline pipeline/environment_function_test.go:98]: You never clear it

File: pipeline/environment_function_test.go (hunk 1 above)
Comment (kubitre) [reply pipeline/environment_function_test.go:46]: why? Using only for using it's no valid way for me.

File: pipeline/environment_function_test.go (hunk 2 above)
Comment (kubitre) [reply pipeline/environment_function_test.go:98]: ok

File: pipeline/environment_function_test.go (hunk 1 above)
Comment (kubitre) [reply pipeline/environment_function_test.go:47]: in this case it is unjustified use, or prove please your view point of using this keyword in this place

Comment (artemkaxboy) [reply pipeline/environment_function_test.go:47]: defer is the only right way to finalize using of resource that must be somehow finalized. No matter how much time will pass after writing the initial code or how many changes you will do, you will never lose the connection between opening any resource and its closing or setting and removing if they are placed right one after another.
What's more, you are removing it in the for loop, it doesn't make any sense. You won't be able to use it in the following iteration.

Comment (artemkaxboy) [reply pipeline/environment_function_test.go:47]: I went to check what's going to happen when you remove an nonexisting env and it turned out that we are using the wrong function. [Remove](https://golang.org/pkg/os/#Remove) removes a file, what we need is [Unsetenv](https://golang.org/pkg/os/#Unsetenv) and it returns error when you removes nonexisting env.

File: lib_test.go [hunk 3]
```diff
@@ -103,3 +112,30 @@ func Test_smartConfigure(t *testing.T) {
...
+	os.Setenv("myField2", "test")
+	os.Setenv("myField3", "true")
+	os.Setenv("myField4", "12.2")
+	os.Setenv("myField5", "true,false,true")
+	defer func() {
+		os.Remove("myField1")
+		os.Remove("myField2")
+		os.Remove("myField3")
+		os.Remove("myField4")
+		os.Remove("myField5")
```
Comment (artemkaxboy) [inline lib_test.go:127]: What you need is os.Unsetenv

File: pipeline/environment_function_test.go [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+	"github.com/goreflect/gostructor/infra"
+	"github.com/goreflect/gostructor/tags"
+)
+
+func TestEnvironmentConfig_GetBaseType(t *testing.T) {
+	myStruct := struct {
+		field1 string `cf_env:"testBaseType"`
+	}{}
+	os.Setenv("testBaseType", "tururu")
+	defer os.Remove("testBaseType")
```
Comment (artemkaxboy) [inline pipeline/environment_function_test.go:18]: What you need is os.Unsetenv

File: pipeline/environment_function_test.go [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+		})
+	}
+}
+
+func TestEnvironmentConfig_GetComplexType(t *testing.T) {
+	myStruct := struct {
+		fieldhard []float32 `cf_env:"mySlice"`
+	}{}
+	os.Setenv("mySlice", "1.12,1.24,1.67")
+	defer os.Remove("mySlice")
```
Comment (artemkaxboy) [inline pipeline/environment_function_test.go:100]: What you need is os.Unsetenv

File: pipeline/environment_function_test.go (hunk 1 above)
Comment (kubitre) [reply pipeline/environment_function_test.go:47]: ++

File: lib_test.go (hunk 3 above)
Comment (kubitre) [reply lib_test.go:127]: +=

File: pipeline/environment_function_test.go (hunk 4 above)
Comment (kubitre) [reply pipeline/environment_function_test.go:18]: +=

File: pipeline/environment_function_test.go (hunk 5 above)
Comment (kubitre) [reply pipeline/environment_function_test.go:100]: ++

**Pull request merged**